mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -648,4 +648,50 @@ mod tests {
 
         assert_eq!(expected_user_agent, user_agent_header_value);
     }
+
+    #[test]
+    #[ignore]
+    // Simple test to ensure the user agent header is correct even when given profile contains calid credentials.
+    fn test_client_with_valid_credentials() {
+        let expected_user_agent = "mountpoint-s3-client";
+        let profile_name_override = String::from("VALID_CREDENTIALS");
+
+        let config = S3ClientConfig {
+            profile_name_override: Some(profile_name_override),
+            ..Default::default()
+        };
+
+        let client = S3CrtClient::new("eu-west-1", config).expect("Create test client");
+
+        let mut message = client
+            .new_request_template("GET", "plutotestankit")
+            .expect("new request template expected");
+
+        let headers = message.inner.get_headers().expect("Expected a block of HTTP headers");
+
+        let user_agent_header = headers
+            .get("User-Agent")
+            .expect("User Agent Header expected with given prefix");
```

Comment:
That's not what we are expecting, only that "user agent header should be set".

```suggestion
        let user_agent_header = headers
            .get("User-Agent")
            .expect("user agent header should be set");
```